admin_scripts/rotate_display.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+#================================================================
+# HEADER
+#================================================================
+#% SYNOPSIS
+#+    rotate_display.sh [normal/right/left]
+#%
+#% DESCRIPTION
+#%    This script looks for all displays connected and rotates the first one from the list.
+#%
+#%    It takes one mandatory parameter. The direction of the rotation.
+#%    Based on xrandr. From xrandr manual:
+#%    "Rotation can be one of 'normal', 'left', 'right' or 'inverted'. This causes the output contents to be
+#%    rotated in the specified direction. 'right' specifies a clockwise rotation of the picture and  'left'
+#%    specifies a counter-clockwise rotation."
+#%
+#================================================================
+#- IMPLEMENTATION
+#-    version         rotate_display.sh (magenta.dk) 1.0.0
+#-    author          Danni Als
+#-    copyright       Copyright 2019, Magenta ApS
+#-    license         GNU General Public License
+#-    email           danni@magenta-aps.dk
+#-
+#================================================================
+#  HISTORY
+#     2019/11/12 : af : Script created
+#     2019/11/14 : af : Changed tactics. Instead of trying to connect to the X-server from root, we now rotate screen upon user login.
+#
+#================================================================
+# END_OF_HEADER
+#================================================================
+
+set -e
+
+if [ "$1" != "normal" -a "$1" != "right" -a "$1" != "left" ]; then
+    echo "Wrong rotation command given: $1"
+    exit -1
+fi
+
+AUTOSTART_FOLDER=/home/.skjult/.config/autostart/
+
+if [ ! -d $AUTOSTART_FOLDER ]; then
+    mkdir $AUTOSTART_FOLDER
+    echo "$AUTOSTART_FOLDER created..."
+fi
+
+ROTATE_SCREEN_FILE=rotate_screen.sh
+ROTATE_SCREEN_FILE_COMPLETE_PATH="$AUTOSTART_FOLDER$ROTATE_SCREEN_FILE"
+
+if [ -f $ROTATE_SCREEN_FILE_COMPLETE_PATH ]; then
+    rm $ROTATE_SCREEN_FILE_COMPLETE_PATH
+    echo "$ROTATE_SCREEN_FILE_COMPLETE_PATH deleted..."
+fi
+cat <<EOT >> "$ROTATE_SCREEN_FILE_COMPLETE_PATH"
```

Comment:
There are two simpler alternatives here: you can either do `rm -f "$ROTATE_SCREEN_FILE_COMPLETE_PATH"` to make sure the file's gone (in a way that doesn't fail if it doesn't exist), or you can just change the redirection operator to `>` to overwrite it.

Reply:
Again I think it is important that we can see in the log what precisely happened so we know which state the filesystem is in...

Reply:
In that case, why not change the `set -e` at the top to `set -xe` to print out every command as it's run? Then the log gives a (more or less) complete view of everything that happened.